internlm/utils/interface.py

```diff
@@ -0,0 +1,135 @@
+import copy
+import warnings
+from dataclasses import dataclass
+from typing import Callable, List, Optional
+
+import torch
+import torch.nn as nn
+from transformers import AutoModel, AutoTokenizer
+from transformers.generation.utils import LogitsProcessorList, StoppingCriteriaList
+from transformers.utils import logging
+
+logger = logging.get_logger(__name__)
+
+
+@dataclass
+class GenerationConfig:
+    max_length: Optional[int] = None
+    top_p: Optional[float] = None
+    temperature: Optional[float] = None
+    do_sample: Optional[bool] = True
+    repetition_penalty: Optional[float] = 1.0
+
+
+@torch.inference_mode()
+def generate_interactive(
```

Comment:
generate_interactive -> generation_iterator()

Reply:
Add some docstring about this function.

Reply:
This function is copied from `web_demo.py`